okrs_2019.md

```diff
@@ -0,0 +1,229 @@
+It often assigns too much cred to [bikesheds] that have lots of activity but produce
+little value.
+
+[bikesheds]: https://en.wikipedia.org/wiki/Law_of_triviality
+
+We're not discouraged by these issues; credit attribution is a hard problem,
+and we're just getting started! In 2019, improving the quality of these
+attributions will be our top priority. Fortunately, we have a lot of ideas on
+how to move forward. Broadly, they fall into two related categories: improving
+the algorithm, and improving the input data.
+
+### Initiatives
+
+#### Algorithm Improvements
+- * [ ] Use temporal data in cred analysis
+  - * [ ] Decide how to represent time (on edges? on nodes?)
+  - * [ ] Decide semantics for time-weighted cred
+  - * [ ] Enables asking "who earned cred this month"?
+  - * [ ] Enables setting explicit weights over different time periods
+- * [ ] Add support for heuristics
+  - * [ ] Define elegant semantics for composition
+  - * [ ] Ensure cohesion with other features (e.g. pass-thru semantics)
+- * [ ] Add 'pass-thru' node semantics ([sourcecred/notes#1])
+- * [ ] Add identity resolution
+- * [ ] Add cred decomposition (who deserves the cred for a specific contribution?)
+
+#### Data Improvements
+- * [ ] Add file-level cred analysis
+- * [ ] Add function call graph cred analysis
+- * [ ] Enable communities to manually add nodes to the graph
+- * [ ] Add heuristics to recognize project stewardship tasks like:
+  - * [ ] writing documentation
+  - * [ ] writing tests
+  - * [ ] triaging issues
+
+These initiatives involve a lot of subtle research and design work. Adding
+heuristics requires designing an algorithm for composing heuristics with good
+mathematical properties, and adding file-level cred means defining an approach
+for consistently representing the identity of an individual file, and keeping
+track of relationships across file renames, splits, and merges.
+
+For these problems, we intend to first write very clear descriptions of the
+problem, relevant considerations, and edge cases. Then, as we progress towards
+a solution, we will document the design rationale as well as alternatives
+considered.
+
+#### Gaming Resistance
+Also, as SourceCred sees more usage, we expect that "gaming" will be an increasing
+problem, as participants try to manipulate the cred graph to inflate their own scores.
+We should document our approach to resisting gaming, explore common attacks,
+and persuasively demonstrate that we can mitigate them.
+
+[sourcecred/notes#1]: https://github.com/sourcecred/notes/issues/1
+
+### Metrics
+
+Coming up with metrics for attribution quality is challenging, since there
+isn't a readily available ground truth to compare against. Right now, the best
+metric we have is asking the contributors to a project to rate the quality of
+our attribution. The "cred rating" in the table below is the average rating
+that contributors to a project give its cred attribution, measured on a
+10-point scale.
```

Comment:
> 1- Meta point: will you stick with "asking contributors" as a best-in-class metric for 2019? You seem to imply it's the least worst, which invites the question "what will be the upgrade" but don't elaborate. A quick "we will keep working off of this metric as ground truth in 2019 but include this section as a reminder of the importance future developments in this area of study will have on SourceCred" would help.

It would be great if we come up with a better metric during 2019, although doing so isn't a top priority right now. If we do come up with a better metric, we'll add it to the doc. (With historical measurements if possible for consistent tracking.) I expect we'll keep cred ratings around at least for all of the year so we can have consistent measurements.

> 2 - Would love a link to a quick doc on the 10-point scale. This seems like it would be a highly highly subjective system, not just on a per-process basis, but based on a contributors' understanding of SourceCred and the underlying project itself (not to mention interpersonal dynamics, politics etc). Are you averaging scores from all contributors to get the "project" score? Are all contributors learning how to rate from the same prompt? Would love to understand better (not in scope for this doc)

Good point. See https://github.com/sourcecred/mission/issues/9 where we can track figuring all of this out.

> 3- Why do you have different goals for SC across both projects? Wouldn't you expect it to perform as well on both?

Updated the doc to explain: 
> We have a higher goal for SourceCred's average rating for two reasons:
> - As a smaller and younger project, attributing its cred is simply easier
> - We think that having a highly engaged community will enable producing better
  attributions, and we expect the SourceCred community to be more engaged

Reply:
This is great!